### PR TITLE
SCB-1422 optimize classmaker

### PR DESCRIPTION
**Pull request by kakulisen**
print the error of the command execution and add the unit test for the ClassMaker

### CODE REVIEW

File: common/src/main/java/org/apache/servicecomb/toolkit/common/ClassMaker.java [hunk 1]
```diff
@@ -17,24 +17,33 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class ClassMaker {
 
   public static void compile(String projectPath) throws IOException, TimeoutException, InterruptedException {
     Runtime runtime = Runtime.getRuntime();
-    Process exec = runtime.exec("mvn clean package -f " + projectPath);
+    String mvnCommand =
+        "mvn clean package -f " + projectPath;
```
Comment (MabinGo) [inline common/src/main/java/org/apache/servicecomb/toolkit/common/ClassMaker.java]: assignment statement could be moved to same line here

File: common/src/test/java/org/apache/servicecomb/toolkit/common/ClassMakerTest.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+    String projectPath = testResourceDirectory + "/projects/demo-with-invalid-pom/";
+
+    try {
+      ClassMaker.compile(projectPath);
+    } catch (RuntimeException e) {
+      Assert.assertThat(e.getMessage(), StringContains.containsString("Command exec fail"));
+      return;
+    }
+    Assert.fail();
```
Comment (MabinGo) [inline common/src/test/java/org/apache/servicecomb/toolkit/common/ClassMakerTest.java]: suggest to add some failure message

File: common/src/test/java/org/apache/servicecomb/toolkit/common/ClassMakerTest.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    ClassMaker.compile(projectPath);
+    Assert.assertTrue(new File(projectPath + "/target").exists());
+  }
+
+  @Test
+  public void runWithInvalidMavenProjectPath() throws InterruptedException, TimeoutException, IOException {
+
+    String projectPath = testResourceDirectory + "/projects/demo-with-invalid-pom/";
+
+    try {
```
Comment (MabinGo) [inline common/src/test/java/org/apache/servicecomb/toolkit/common/ClassMakerTest.java]: suggest to optimize code as below,
try {
      ClassMaker.compile(projectPath);
      Assert.fail("error message");
    } catch (RuntimeException e) {
      Assert.assertThat(e.getMessage(), StringContains.containsString("Command exec fail"));
    }

File: common/src/test/java/org/apache/servicecomb/toolkit/common/ClassMakerTest.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+import java.io.IOException;
+import java.util.concurrent.TimeoutException;
+
+import org.hamcrest.core.StringContains;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class ClassMakerTest {
+
+  private static final String testResourceDirectory = "./src/test/resources";
```
Comment (MabinGo) [inline common/src/test/java/org/apache/servicecomb/toolkit/common/ClassMakerTest.java]: it's better to use File.separator instead /

**Comment by kakulisen** [comment]
Thanks for your suggestion.  I have revised it according to the suggestion，please help to check again

**Comment by MabinGo** [comment]
LGTM

**Pull request merged**